cni-plugin/main.go

```diff
@@ -98,6 +99,24 @@ func configureLoggingLevel(logLevel string) {
 	}
 }
 
+// A debug log is only intended for development of linkerd-cni.
+//
+// Linkerd-cni does not manage log rotation of this file, administrators
+// who choose to enable this for production use SHOULD provide log rotation.
+//
+// Use a MultiWriter to write both to stderr and to a logfile if
+// the environment variable LINKERD_CNI_DEBUG_LOGFILE is passed in.
+func setupDebugLog(pod *v1.Pod) {
+	if filename := pod.GetObjectMeta().GetAnnotations()["LINKERD_CNI_DEBUG_LOGFILE"]; filename != "" {
+		f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0644) //nolint:gosec
+		if err != nil {
+			panic(fmt.Sprintf("failed to create debug log file at %s with err %e", filename, err))
```

Comment:
If we end up doing it this way we might want to rethink this panic. It made sense when we were setting everything up, but I don't think we should panic in the middle of the pod processing.

For example, if in a more restricted environment we cannot create a debug fail, we would end up failing the whole operation -- side effect probably outweighs the good this function would do.